### PR TITLE
User/alkalomo/prepare one box only for main

### PR DESCRIPTION
**Pull request by alkalomo**
With this change we introduce the following:
- improved vocabulary for the different operations performed by the tool
- Enhanced UserConfig that can handle multiple scale units with workloads deployed across scale units
- OneBoxOnly capability (more than one scale unit deployed on the same environment)
- BugFix to restore backwards compatibility with 10.0.16

### CODE REVIEW

**Review by cabeln — APPROVED** [review]

File: src/ScaleUnitManagement/ScaleUnitFeatureManager/Common/StopServices.cs [hunk 1]
```diff
@@ -1,31 +1,47 @@
...
-            ce.RunCommand(cmd);
+
+            try
+            {
+                ce.RunCommand(cmd);
+            }
+            catch (Exception e)
+            {
+                if (!Config.UseSingleOneBox())
+                    throw e;
```
Comment (AndreasHassing) [inline src/ScaleUnitManagement/ScaleUnitFeatureManager/Common/StopServices.cs:43]: Just `throw;` or you lose the original call-stack 🙂.

File: src/ScaleUnitManagement/UserConfig.sample.xml [hunk 2]
```diff
@@ -1,46 +1,73 @@
 ﻿<?xml version="1.0" encoding="utf-8" ?>
 <CloudAndEdgeConfiguration>
-  <!--EnivronmentType represents the infrastructure you host on.
-      It should be VHD if you have OneBox or LCSVHD.
-      It should be LCSHosted if you have an LCS cloud-hosted devbox.-->
-  <EnvironmentType>LCSHosted</EnvironmentType>
-  <AxDbName>AxDB</AxDbName>
-
+  
+  <!-- Use this flag to indicate whether you want to use on single on-box to host both teh hub and the scale unit -->
```
Comment (AndreasHassing) [inline src/ScaleUnitManagement/UserConfig.sample.xml:4]: ... to use one\* single one\*-box to host both the\* hub ...

File: src/ScaleUnitManagement/UserConfig.sample.xml [hunk 3]
```diff
@@ -1,46 +1,73 @@
...
+    <!-- Specify the URL of the authority endpoint of your tenant-->
     <Authority>https://login.windows.net/YOUR_AAD_TENANT</Authority>
+    
     <!-- If empty, the existing value in web.config will be used - otherwise the web.config will get overwritten with this value. -->
     <AADTenantId></AADTenantId>
-    <AzureStorageConnectionString></AzureStorageConnectionString>
   </AADConfiguration>
-
+  
+  <!-- This sections lists your environment, beeing the hub and your scale unit.-->
```
Comment (AndreasHassing) [inline src/ScaleUnitManagement/UserConfig.sample.xml:27]: This section* lists your environments, one being the hub and any other being scale units.

File: src/ScaleUnitManagement/Utilities/AxDeployment.cs [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+            return Instance().ApplicationVersion;
+        }
+
+        public static bool IsApplicationVersionMoreRecentThan(string version)
+        {
+            string[] deployedApplicationVersion = GetApplicationVersion().Split('.');
+            string[] compareToVersion = version.Split('.');
+
+            for (int i = 0; i < deployedApplicationVersion.Length && i < compareToVersion.Length; i++)
```
Comment (AndreasHassing) [inline src/ScaleUnitManagement/Utilities/AxDeployment.cs:42]: With this logic, it says that 10.0.16.5532 is more recent than than 10.0.15.6373 - can that occur?

Comment (AndreasHassing) [reply src/ScaleUnitManagement/Utilities/AxDeployment.cs:42]: Nvm I see tests below.

**Review by AndreasHassing — APPROVED** [review]
Approving with suggestions. Nice work @garoymsft & @alkalomo.

**Pull request merged**